herzog/__init__.py

```diff
@@ -38,10 +39,51 @@ def __exit__(self, *args, **kwargs):
                     del __main__.__dict__[key]
             __main__.__dict__.update(self._state_modules)
 
-def generate(handle: TextIO):
-    cells = [obj.to_ipynb_cell() for obj in parse_cells(handle)
+def load_ipynb_cells(ipynb: TextIO) -> List[Dict[Any, Any]]:
+    try:
+        cells = json.loads(ipynb.read()).get('cells', None)
+    except Exception:
```

Comment:
I figured any error opening the file and reading it would indicate that it wasn't a ipynb file, though I guess it's a little broad.  Will change to `JSONDecodeError`.